hydromt/models/model.py

```diff
@@ -121,10 +118,10 @@ def __init__(
         # file system
         self.root: ModelRoot = ModelRoot(root or ".", mode=mode)
 
-        self._components: Dict[str, ModelComponent] = {}
+        self._components: dict[str, ModelComponent] = {}
```

Comment:
I thought we needed to use typing and not python types?

Reply:
since python 3.9 they are equivalent, Though I personally prefer using the Typing library since I think it reads a bit better. I think it's mostly due to convention though. We haven't made an agreement on this yet. Do others have a preference?

Reply:
I'm okay using Typing library.